src/app/payment/paymentMethod/MasterpassPaymentMethod.tsx

```diff
@@ -20,8 +20,17 @@ const MasterpassPaymentMethod: FunctionComponent<MasterpassPaymentMethodProps &
         },
     }), [initializePayment]);
 
+    const getLocale = useCallback((browserLanguage: string) => {
+        const localePayload: string = browserLanguage.includes('-') ? browserLanguage.replace('-', '_') : `${browserLanguage}_${browserLanguage}`;
+
+        return localePayload.toLowerCase();
+    }, []);
+
     const { config: { testMode }, initializationData: { checkoutId, isMasterpassSrcEnabled } } = rest.method;
-    const locale = navigator.language.replace('-', '_');
+    const { language: localeLanguage } = navigator;
+
+    const locale = getLocale(localeLanguage);
```

Comment:
The current language in checkout might not match the one from the browser. Would it be better to honour the one set for the rest of storefront in this case?